### PR TITLE
Remove mentions of our fork of `go-ethereum` from docs.

### PR DESCRIPTION
**Pull request by mandrigin**
Cleaning-up leftovers.

### CODE REVIEW

**Review by dshulyak — APPROVED** [review]

File: _assets/docs/DEPENDENCIES.md [hunk 1]
```diff
@@ -49,10 +49,10 @@ file is changed, `dep ensure` will re-generate `Gopkg.lock` as well.
 
 ## Updating `Geth`
 
-1. Update `develop` branch in [`status-im/go-ethereum`](https://github.com/status-im/go-ethereum/tree/develop).
-2. Update the `go-ethereum` dependency: `dep ensure -v -update github.com/ethereum/go-ethereum`.
-3. Make sure that `[[constraint]]` statements in `status-go/Gopkg.toml` contains the same SHAs as `go-ethereum/vendor/vendor.json`.
-4. Update vendor files in `status-go`, running `dep ensure`.
+1. Update the `go-ethereum` dependency: `dep ensure -v -update github.com/ethereum/go-ethereum`.
```
Comment (pedropombeiro) [inline _assets/docs/DEPENDENCIES.md]: A bit unrelated, but what do you think about making a `make update-geth` target? Also, updating geth to a new release would involve editing the branch name in `Gopkg.toml`, right?

File: _assets/docs/DEPENDENCIES.md [hunk 2]
```diff
@@ -49,10 +49,10 @@ file is changed, `dep ensure` will re-generate `Gopkg.lock` as well.
...
+2. Apply patches using [the patcher script](../patches/patcher). Update patches to resolve conflicts if needed.
+3. Update vendor files in `status-go`, running `dep ensure`.
```
Comment (PombeirP) [inline _assets/docs/DEPENDENCIES.md]: Do we want to reference instead `make dep-ensure`?

Comment (mandrigin) [reply _assets/docs/DEPENDENCIES.md]: agree!

File: _assets/docs/DEPENDENCIES.md (hunk 1 above)
Comment (mandrigin) [reply _assets/docs/DEPENDENCIES.md]: > updating geth to a new release would involve editing the branch name in Gopkg.toml

only for major releases, but I'll add it to the doc

**Review by pedropombeiro — APPROVED** [review]

**Pull request merged**